Add Only update vanadium checkbox to EngDiff calibration tab

DanielMurphy22 opened the pull request:
**Description of work.**

Add OnlyUpdateVanadium checkbox to skip calibration if checked
    
Only_Update_vanadium checkbox is enabled if Create_New_calibration is enabled
and the last calibration was successful.
When Only_update_vanadium is checked, crop is disabled and unchecked,
Sample path is disabled, its path not validated,
and the Main action button is renamed to "Update Vanadium"


**To test:**

- run `git test-pr origin 32184`
- Open Workbench and the Engineering Diffraction interface

_Enabling_
- The Only Update Vanadium checkbox should be disabled (greyed out)
- For Vanadium Path enter: .`../cycle_19_1/ENGINX00307521.nxs` or your favourite dataset
- For Sample Path enter: `.../cycle_19_1/ENGINX00305738.nxs` or your favourite dataset
- make sure Plot Calibrated workspace is ticked
- Then click Calibrate
- Screenshot or just keep open the plots that appear
- After it runs successfully, the Only Update Vanadium checkbox should be enabled

_Ticking/Checking_
- Tick Crop Calibration
- Now tick Only Update Vanadium and notice that this has:
    - Disabled Sample Path
    - Disabled and unticked Crop Calibration
    - The main action button is renamed from "Calibrate" to "Update Vanadium"
- The enabled/disabled state of these inputs persist after toggling to Load Existing Calibration and then toggling back to Create New Calibration

_Validation and Behaviour_
- Firstly, untick OnlyUpdateVanadium, and change the Sample path to another dataset, such as: `.../cycle_12_4/ENGINX00193749.nxs`
- Now retick Only Update Vanadium and click the main action button "Update Vanadium"
- The output plots should be the same as before!!
- Now untick OnlyUpdateVanadium and click Calibrate
- The newly output plot should be different, which proves that the sample path was ignored when OnlyUpdateVanadium was ticked.

Questions for tester:
- **Should the calibration be replotted if Only_update_vanadium is ticked? (calibration should not change!)** -> I've decided Yes!
- Are there any other tests that should be added?

Fixes #31904 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

DanielMurphy22 committed:
Skip calibration if update_vanadium checked

Only_Update_vanadium checkbox is enabled if Create_New_calib is enabled
and the last calibration was successful.
When Only_update_vanadium is checked, crop is disabled and unchecked,
Sample path is disabled, its path not validated,
and the Main action button is renamed to "Update Vanadium"

DanielMurphy22 committed:
Update test_calib_presenter for OnlyUpdateVanadium checkbox

Included new only_update_vanadium param for the worker to pass to model
Added two tests to check that when only_update_vanadium is ticked,
an invalid sample path doesn't stop the workflow, but an invalid
vanadium path still does block progress to the model.
Also added release notes

DanielMurphy22 committed:
Emit signal of successful calibration to OnlyUpdateVan checkbox

Update test_calib_presenter

DannyHindson reviewed (commented):
Nice work on the UI aspect of this. The enabling\disabling of the new checkbox and the change of label on the button means this is more simple than I'd feared.

I've found an issue after running the "Only update Vanadium" option after first having created a cropped calibration. So I did:

1. Create New Calibration with Crop Calibration set to North Bank
2. Checked the "Only update Vanadium" checkbox
3. Clicked on the Update Vanadium button

...and then I get this error:

![image](https://user-images.githubusercontent.com/56295817/129061872-23a23e2c-198e-4f0a-81cd-290b83df6ac6.png)

Looks like the code is expecting some South Bank workspaces to exist that aren't there.

DanielMurphy22 committed:
Do not plot calibration if Only_update_vanadium ticked

DanielMurphy22 commented:
> I've found an issue after running the "Only update Vanadium" option after first having created a cropped calibration. 

Great spot! I hadn't thought about this. Its caused by trying to replot the calibration when only updating the vanadium and obviously in this case me undoing the crop input has meant that the plotting code is expecting  non-cropped data! I think the simplest solution is to not replot the calibration if OnlyUpdateVanadium is ticked ( as I was considering anyway in my question in the PR description, so this works well!)

DannyHindson commented on scripts/Engineering/gui/engineering_diffraction/tabs/calibration/model.py:
```diff
@@ -44,60 +45,62 @@ def create_new_calibration(self,
         # vanadium corrections workspaces not used at this stage, but ensure they exist and create if not
         vanadium_corrections.fetch_correction_workspaces(vanadium_path, instrument, rb_num=rb_num)
-        ceria_workspace = path_handling.load_workspace(ceria_path)
         full_calib_path = get_setting(output_settings.INTERFACES_SETTINGS_GROUP,
```
It seems like these ~7 lines relating to full_calib_path could also be moved inside the `if not only_update_vanadium` statement? I guess they also result in a workspace being loaded as well as populating the full_calib_path variable so we also need to consider if that workspace being present is an important part of the "Only update Vanadium" process. It seems like it's not to me because the user will always have either created\loaded a calibration prior to "Only update Vanadium"

DannyHindson reviewed (changes requested):
I've added one detail code comment. I've got two more general comments about the workflow:

- since you're not plotting the calibration results any more when doing the "Only update Vanadium" action I think the "Plot Calibrated Workspace" checkbox needs disabling when the "Only update Vanadium" checkbox is ticked

- if I have "Force Vanadium Recalculation" set to False in the settings and I run a "Only update Vanadium" action I think the UI just loads any pre-existing V files if they exist. This seems a bit counterintuitive.

The 2nd point might not be v easy to fix though and might benefit from a chat with Richard when he's back. Let's have a chat in a bit and see what you think

DanielMurphy22 committed:
Disable Plot checkbox when OnlyUpdateVanadium checked

Also made sure that a calibration completing succssfully does not incorrectly
re-enable the plot checkbox
Update Unit tests

DannyHindson committed:
Fix problem with enabling of Only Update Vanadium

If you open the engineering UI without a last calibration being
stored in the workbench ini file, the Only Update Vanadium checkbox
was enabled even though no calibration had been previously run or
loaded.
Fixed this by calling set_create_new_enabled in presenter constructor
Also remove unnecessary call to set the button text to Calibrate

DannyHindson committed:
Create separate load vanadium section

Create separate load vanadium section on the calibration
tab and remove the force vanadium option from settings
and only update vanadium checkbox

DannyHindson commented:
This PR has been superseded by https://github.com/mantidproject/mantid/pull/32229
If I get time I will back out the last couple of commits so we have a record of the alternative (original) solution in case we get some feedback that the solution on https://github.com/mantidproject/mantid/pull/32229 isn't OK

DannyHindson commented:
Closing now that this has been superseded
